# HG changeset patch
# User Jochen Bartl <jochen.bartl@gmail.com>
# Date 1388345633 -3600
#      Sun Dec 29 20:33:53 2013 +0100
# Node ID 7e9ac37a75afc44d82fbeb871abc0aba62bc2c1f
# Parent  4c0656fde448f46c9c0a8d729d99d2eccdad4379
Implements a layer for the VXLAN protocol

diff -r 4c0656fde448 -r 7e9ac37a75af scapy/contrib/vxlan.py
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/scapy/contrib/vxlan.py	Sun Dec 29 20:33:53 2013 +0100
@@ -0,0 +1,21 @@
+# Virtual eXtensible Local Area Network (VXLAN)
+# https://tools.ietf.org/html/draft-mahalingam-dutt-dcops-vxlan-06
+
+# scapy.contrib.description = VXLAN
+# scapy.contrib.status = loads
+
+from scapy.packet import *
+from scapy.fields import *
+
+class VXLAN(Packet):
+    name = "VXLAN"
+    fields_desc = [ FlagsField("flags", 0x08, 8, ['R', 'R', 'R', 'I', 'R', 'R', 'R', 'R']),
+                    X3BytesField("reserved1", 0x000000),
+                    X3BytesField("vni", 0x000000),
+                    XByteField("reserved2", 0x00)]
+
+    def mysummary(self):
+        return self.sprintf("VXLAN (vni=%VXLAN.vni%)")
+
+bind_layers(UDP, VXLAN, dport=4789)
+bind_layers(VXLAN, Ether)
# HG changeset patch
# User Jochen Bartl <jochen.bartl@gmail.com>
# Date 1389135944 -3600
#      Wed Jan 08 00:05:44 2014 +0100
# Node ID e462cd86539ed2c4877499f7718c1dabb9e1498c
# Parent  7e9ac37a75afc44d82fbeb871abc0aba62bc2c1f
Added tests for the VXLAN layer

diff -r 7e9ac37a75af -r e462cd86539e scapy/contrib/vxlan.utscapy
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/scapy/contrib/vxlan.utscapy	Wed Jan 08 00:05:44 2014 +0100
@@ -0,0 +1,8 @@
+% VXLAN Tests
+* Tests for the Scapy VXLAN layer
+
+= Build a VXLAN packet with VNI of 42
+str(UDP(sport=1024, dport=4789, len=None, chksum=None)/VXLAN(flags=0x08, vni=42)) == "\x04\x00\x12\xb5\x00\x10\x00\x00\x08\x00\x00\x00\x00\x00\x2a\x00"
+
+= Verify VXLAN Ethernet Binding
+str(VXLAN(vni=23)/Ether(dst="11:11:11:11:11:11", src="11:11:11:11:11:11", type=0x800)) == "\x08\x00\x00\x00\x00\x00\x17\x00\x11\x11\x11\x11\x11\x11\x11\x11\x11\x11\x11\x11\x08\x00"
